pkg/trace/test/testsuite/events_test.go

```diff
@@ -0,0 +1,98 @@
+package testsuite
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/trace/pb"
+	"github.com/DataDog/datadog-agent/pkg/trace/test"
+)
+
+func jsonTraceFromPath(path string) (pb.Trace, error) {
+	slurp, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+	var t pb.Trace
+	if err := json.Unmarshal(slurp, &t); err != nil {
+		return t, err
+	}
+	return t, err
+}
+
+func TestAPMEvents(t *testing.T) {
+	runner := test.Runner{Verbose: true}
+	if err := runner.Start(); err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
+		if err := runner.Shutdown(time.Second); err != nil {
+			t.Log("shutdown: ", err)
+		}
+	}()
+
+	traceWithRates, err := jsonTraceFromPath("./testdata/trace_with_rates.json")
+	if err != nil {
+		t.Fatal(err)
+	}
+	traceWithoutRates, err := jsonTraceFromPath("./testdata/trace_without_rates.json")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	t.Run("off", func(t *testing.T) {
+		if err := runner.RunAgent(nil); err != nil {
+			t.Fatal(err)
+		}
+		defer runner.KillAgent()
+
+		if err := runner.Post(pb.Traces{traceWithoutRates}); err != nil {
+			t.Fatal(err)
+		}
+
+		waitForTrace(t, &runner, func(v pb.TracePayload) {
+			if n := len(v.Transactions); n != 0 {
+				t.Fatalf("expected no events, got %d", n)
+			}
+		})
+	})
+
+	t.Run("env", func(t *testing.T) {
+		os.Setenv("DD_APM_ANALYZED_SPANS", "coffee-house|servlet.request=1")
+		defer os.Unsetenv("DD_APM_ANALYZED_SPANS")
+		if err := runner.RunAgent(nil); err != nil {
+			t.Fatal(err)
+		}
+		defer runner.KillAgent()
+
+		if err := runner.Post(pb.Traces{traceWithoutRates}); err != nil {
+			t.Fatal(err)
+		}
+
+		waitForTrace(t, &runner, func(v pb.TracePayload) {
+			if n := len(v.Transactions); n != 1 {
+				t.Fatalf("expected 1 event, got %d", n)
+			}
+		})
+	})
+
+	t.Run("client", func(t *testing.T) {
+		if err := runner.RunAgent(nil); err != nil {
+			t.Fatal(err)
+		}
+		defer runner.KillAgent()
+
+		if err := runner.Post(pb.Traces{traceWithRates}); err != nil {
+			t.Fatal(err)
+		}
+
+		waitForTrace(t, &runner, func(v pb.TracePayload) {
```

Comment:
Nit: given that we have `runner.Post` and other methods, wouldn't it be better (from a consistency point of view) to have `runner.WaitForTrace` as well?

Reply:
It did cross my mind, but I figured best keep that package slim and focused, as this can always be just a test helper. It's also somewhat test specific. If we start doing that, it will just end up in multiple forms (e.g. "wait for stats", "wait for services", etc.) complicating things. 🦐 

I find it somewhat confusing anyway that you have to instantiate a `test.Runner`, then `Start` it (what are we starting?), then call `RunAgent` on it, then `Post`. Even this part I'd like to simplify but I don't have better ideas. Luckily I wrote an example which refreshes my memory when I consult it.